src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -388,17 +388,28 @@ protected void LogEventsFromTextOutput (string singleLine, MessageImportance mes
 				if (message.StartsWith ("error: ", StringComparison.InvariantCultureIgnoreCase))
 					message = message.Substring ("error: ".Length);
 
-				LogError ("APT0000", message, file, line);
-				return;
+				if (level.Contains ("error") || (line != 0 && !string.IsNullOrEmpty (file))) {
+					LogError ("APT0000", message, file, line);
+					return;
+				}
 			}
 
-			// Handle additional error that doesn't match the regex
-			if (singleLine.Trim ().StartsWith ("invalid resource directory name:")) {
-				LogError ("APT0000", string.Format ("Invalid resource directory name: \"{0}\".", singleLine.Substring (singleLine.LastIndexOfAny (new char[] { '\\', '/' }) + 1)), ToolName);
-				return;
+			// Handle additional errors that doesn't match the regex
+			foreach (var knownInvalidError in KnownInvalidErrorMessages) {
+				if (singleLine.Trim ().StartsWith (knownInvalidError, StringComparison.OrdinalIgnoreCase)) {
+					LogError ("APT0000", string.Format ("{0} \"{1}\".", knownInvalidError, singleLine.Substring (singleLine.LastIndexOfAny (new char [] { '\\', '/' }) + 1)), ToolName);
```

Comment:
Why do we only provide the "rightmost" directory name, and not the entire path? Is it too long and noisy?

Reply:
no idea. its been that way for a long time from what I can see.